src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -253,6 +285,12 @@ impl<'a, 'tcx, 'rcx, 'cstore> DocFolder for LinkCollector<'a, 'tcx, 'rcx, 'cstor
         let cx = self.cx;
         let dox = item.attrs.collapsed_doc_value().unwrap_or_else(String::new);
 
+        look_for_tests(&cx, &dox, &item);
+
+        if !UnstableFeatures::from_environment().is_nightly_build() {
+            return None;
+        }
```

Comment:
Would it be possible to cache `UnstableFeatures::from_environment()` so we don't have to poke environment variables for every item in the crate?

Reply:
Sure, I'll put it in another commit though.